### PR TITLE
LibJS: break or continue with nonexistent label is a syntax error

### LINKED ISSUE #3714: LibJS: break/throw statement with nonexistent label should be a syntax error
Serenity:
```js
> for (let i = 0; i < 10; ++i) { continue loop; }
undefined
> for (let i = 0; i < 10; ++i) { break loop; }
undefined
```

Spidermonkey: 
```js
> for (let i = 0; i < 10; ++i) { continue loop; }
Uncaught SyntaxError: label not found
> for (let i = 0; i < 10; ++i) { break loop; }
Uncaught SyntaxError: label not found
```

### PR DESCRIPTION
**Pull request by mattco98**
Fixes #3714.

### CODE REVIEW

File: AK/StringView.h [hunk 1]
```diff
@@ -187,6 +187,11 @@ class StringView {
     size_t m_length { 0 };
 };
 
+template<>
+struct Traits<StringView> : public GenericTraits<String> {
+    static unsigned hash(const StringView& s) { return s.hash(); }
+};
+
```
Comment (linusg) [inline AK/StringView.h:194]: Let's maybe put this in a separate commit?!

**Comment by awesomekling** [comment]
Note that using `StringView` as a hash key may be less optimal than `String` since `String` will cache its hash while `StringView` may have to recompute it every time it's requested :)

**Comment by mattco98** [comment]
Ahh interesting, didn't know that! I was just trying to avoid the `String` copies.
